clang/test/3C/processor.py

```diff
@@ -1,8 +1,11 @@
+#!/usr/bin/env python
 import fileinput 
 import sys
 import os
 
-path_to_monorepo = "/Users/shilpa-roy/checkedc-clang/build/bin/"
+os.chdir(os.path.dirname(__file__))
+# Relative path from clang/test/3C/
+bin_path = "../../../build/bin/"
```

Comment:
Not everyone puts their build directory here. I guess that's better than the hardcoded one you removed; is there a way to just inherit from the current PATH instead?

Reply:
I could do that.  But do we want to require everyone to set their PATH to include the build directory of the current working tree?  In the work I've done so far, I've rarely had a need to set it and thus haven't, though it's possible that will change as I start actually developing the program analysis.  Remember that users may have many working trees (at minimum, one each for 3C and 5C, but I'm already using a lot more, and g5c as currently planned will encourage that too), so we'll need an easy way for users to set the proper environment variables when they start working on a given working tree, sort of like how [activating a Python virtual environment](https://docs.python.org/3/tutorial/venv.html) works now.

What are the use cases for putting the build directory in a nonstandard location?  As long as there is still only one build directory per working tree, the user can always create a symlink from the standard `build` location to the custom location.  ISTM the really difficult case is when there are multiple build directories for the same working tree, e.g., to test multiple sets of build flags.  And ISTM the user could alternatively use a separate working tree per build directory that they want, at the cost of having to propagate source code changes back and forth every time they want to test.  So we have to weigh the value to a user of supporting multiple build directories for the same working tree (compared to having them use multiple working trees) against the complexity it will add to all of our development tools and processes.

We can always check both an environment variable and the standard location (TBD the priority between those two).  My preference would be to avoid assuming that just because we found `3c`, etc. in the PATH, they are the correct ones (I know `lit` tries to avoid this), and recognize only a purpose-specific variable (just `LLVM_OBJ`?), but I could be convinced that this isn't an important enough problem in practice.

Perhaps we should do one (simple) thing in this PR and open a follow-up issue to make a decision on the larger design problem and change all our tools accordingly.  Probably that "one thing" should be consistent with the precedent of our other existing tools; if no such precedent exists, I would prefer to assume the standard build directory location (as in the current PR) but could be convinced otherwise.  For example, it looks like `clang/tools/3c/utils/convert-commands.py` actually requires `LLVM_OBJ` to be set in _all_ cases, though I'm not sure that is intentional: if `LLVM_OBJ` is not set, it falls back to trying to execute an empty string as the file path of `3c` (which will never work), whereas maybe the intent was to execute `3c` and assume it is in the PATH.  @sroy4899 or @Machiry do you know the intent here?

Reply:
This is a lot of "ink" for such a small issue. 

To me it feels ugly that to run the script you need to potentially edit it to set the path. 

I don't care to get into a long debate on the topic, so let's do this: Edit the start of the script to see if the needed executables are located via the default `bin_path.` If they are not, try looking for them in the `PATH`. If you still can't find them, fail with a specific error.

Reply:
> This is a lot of "ink" for such a small issue.

Maybe at this point.  You know I like to get such things right for the long run. :/

> I don't care to get into a long debate on the topic, so let's do this: Edit the start of the script to see if the needed executables are located via the default `bin_path.` If they are not, try looking for them in the `PATH`. If you still can't find them, fail with a specific error.

I decided to improve this a little further: check `$LLVM_OBJ`, then the standard build directory location, then `$PATH`, otherwise fail with a specific error.  Then the only case in which we silently do the wrong thing is if `$PATH` has a copy of `3c` that the user doesn't want to use and `3c` isn't found the other ways.  OK?

Reply:
OK, sounds good to me!